tapchannel/aux_leaf_signer.go

```diff
@@ -240,36 +240,53 @@ func (s *AuxLeafSigner) processAuxSigBatch(chanState *channeldb.OpenChannel,
 
 	defer s.Wg.Done()
 
+	shutdownErr := fmt.Errorf("tapd is shutting down")
 	log.Tracef("Processing %d aux sig jobs", len(sigJobs))
 
 	for idx := range sigJobs {
 		sigJob := sigJobs[idx]
 		cancelAndErr := func(err error) {
 			log.Errorf("Error processing aux sig job: %v", err)
 
-			close(sigJob.Cancel)
+			// Check that the cancel signal was not already sent
+			// before cancelling all jobs. The cancel signal must
+			// only be sent exactly once.
+			select {
+			case <-sigJob.Cancel:
+			default:
+				close(sigJob.Cancel)
+			}
+
 			sigJob.Resp <- lnwallet.AuxSigJobResp{
 				Err: err,
 			}
 		}
 
-		// If we're shutting down, we cancel the job and return.
+		// Check for cancel or quit signals before beginning the job.
 		select {
+		case <-sigJob.Cancel:
+			continue
 		case <-s.Quit:
-			cancelAndErr(fmt.Errorf("tapd is shutting down"))
+			cancelAndErr(shutdownErr)
 			return
-
 		default:
 		}
 
 		// If there is no commit blob, this isn't a custom channel. We
 		// still need to signal the job as done though, even if we don't
 		// have a signature to return.
 		if sigJob.CommitBlob.IsNone() {
-			sigJob.Resp <- lnwallet.AuxSigJobResp{
```

Comment:
So did we ever figure out for certain, why in the trace a goroutine could be blocked here, assuming the channel is always buffered?

Reply:
No, I imagine it's just a forced shutdown that happened to be when this send was happening. I double checked and that `Resp` chan is definitely constructed as buffered.